engine/src/cmd/unit_json_factory.cpp

```diff
@@ -24,16 +24,12 @@
 #include "unit_json_factory.h"
 #include "unit_csv_factory.h"
 
-#include <iostream>
 #include <fstream>
-#include <sstream>
 #include <vector>
 #include <string>
 #include <map>
 #include <boost/json.hpp>
 
-#include "resource/json_utils.h"
-
 
 void UnitJSONFactory::ParseJSON(VSFileSystem::VSFile &file, bool player_ship) {
```

Comment:
My code (deficiencies and all) assumed the following:
```
[
   {"a": "1", "b": 2},
   {"a": "1", "b": 2},
   {"a": "1", "b": 2},
   ...
]
```

However, this looks like it supports other file formats such as `{"a": "1", "b": 2}` without an array. Why would we do this?

Reply:
Because that's how the Boost JSON serialization function constructs it. So when we go to load the file, we need to support either format -- with or without an array at the top level.

Reply:
To clarify further: Roy, your code saved an array with a single object element inside of it:

```json
[
    {
        ...
    }
]
```

My code simply removes the root array:

```json
{
    ...
}
```

Reply:
Unfortunately, we actually want the array. Right now the game saves each ship in its own file. There's no reason to do this. We just need to figure the whole player_ship issue. Either, replace it with player_ship_1...2...3 or player_ship_<random 6 alphanumeric> or something like that and check for prefix.

This would also solve the issue of not supporting a fleet containing two ships of the same type.

Reply:
I see.